verifiers/internal/gcb/verifier.go

```diff
@@ -40,9 +41,70 @@ func (v *GCBVerifier) VerifyArtifact(ctx context.Context,
 
 // VerifyImage verifies provenance for an OCI image.
 func (v *GCBVerifier) VerifyImage(ctx context.Context,
-	artifactImage string,
+	provenance []byte, artifactImage string,
 	provenanceOpts *options.ProvenanceOpts,
 	builderOpts *options.BuilderOpts,
 ) ([]byte, string, error) {
-	return nil, "todo", serrors.ErrorNotSupported
+	prov, err := ProvenanceFromBytes(provenance)
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Verify signature on the intoto attestation.
+	if err = prov.VerifySignature(); err != nil {
+		return nil, "", err
+	}
+
+	// Verify intoto header.
+	if err = prov.VerifyIntotoHeaders(); err != nil {
+		return nil, "", err
+	}
+
+	// Verify the builder.
+	builderID, err := prov.VerifyBuilderID(builderOpts)
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Verify subject digest.
+	if err = prov.VerifySubjectDigest(provenanceOpts.ExpectedDigest); err != nil {
+		return nil, "", err
+	}
+
+	// Verify source.
+	if err = prov.VerifySourceURI(provenanceOpts.ExpectedSourceURI); err != nil {
+		return nil, "", err
+	}
+
+	// Verify branch.
+	if provenanceOpts.ExpectedBranch != nil {
+		if err = prov.VerifyBranch(*provenanceOpts.ExpectedBranch); err != nil {
+			return nil, "", err
+		}
+	}
+
+	// Verify the tag.
+	if provenanceOpts.ExpectedTag != nil {
+		if err := prov.VerifyTag(*provenanceOpts.ExpectedTag); err != nil {
+			return nil, "", err
+		}
+	}
+
+	// Verify the versioned tag.
+	if provenanceOpts.ExpectedVersionedTag != nil {
+		if err := prov.VerifyVersionedTag(*provenanceOpts.ExpectedVersionedTag); err != nil {
+			return nil, "", err
+		}
+	}
+
+	// TODO: verify summary information:
```

Comment:
ah you have imagesummary verification here :)

Reply:
yes. Plus the human-readable part if possible, because that's the part users may see by opening the file.